ShellCheck reports wrong URL for installation in the  Run Shellcheck Task

<!-- Fill in the below form so that we have the relevant details about the environment where the bug/error is occurring. -->
## Environment Details
* **ShellCheck Task**
    <!-- Check the box to indicate which of the ShellCheck Pipeline Tasks you are experiencing an issue with. -->
  * [ ] Install ShellCheck
    * Version: 0.1.16
  * [x] Run ShellCheck
      * Version: 0.3.16
* **Agent Type**
    <!-- Check the box to indicate which Azure Pipelines Agent environment(s) where the error occurs. -->
  * [x] Hosted Linux Agent
  * [ ] Hosted MacOS Agent
  * [ ] Hosted Windows Agent
  * [ ] Self-hosted/private Agent
* **Agent Tool Versions**
    <!-- Add the versions of the below items on the agent where you are seeing the bug. You may include multiple values (i.e. Node 8.x, 9.x) if you are seeing the error on multiple versions. -->
  * **Node.js Version**: Node.js (v12.16.1)
  * **npm version**: according to https://github.com/actions/virtual-environments/blob/master/images/linux/Ubuntu1604-README.md there is no npm on the machine

## Description
<!-- Provide a clear and concise description of the bug/problem you are experiencing. -->
   The Run Task trhows an error saying that shellcheck was installed from the wrong location.
## Log Output
<!-- Provide the log output of the ShellCheck task(s) from your pipeline with the error details. Below is an example of what that will look like: -->
```
Starting: Install ShellCheck
==============================================================================
Task         : Install ShellCheck
Description  : Installs Specified ShellCheck Version
Version      : 0.1.16
Author       : Swellaby
Help         : [More Information](https://github.com/swellaby/azdo-shellcheck)
==============================================================================
Downloading: https://shellcheck.storage.googleapis.com/shellcheck-latest.linux.x86_64.tar.xz
/bin/tar xC /home/vsts/work/_temp/shellcheck-latest -f /home/vsts/work/_temp/a73463fe-8e7c-4efd-a92a-60ed4f18c8d4
Prepending PATH environment variable with directory: /home/vsts/work/_temp/shellcheck-latest/shellcheck-latest
Finishing: Install ShellCheck

Starting: Run ShellCheck
==============================================================================
Task         : ShellCheck
Description  : Run ShellCheck
Version      : 0.3.16
Author       : Swellaby
Help         : [More Information](https://github.com/swellaby/azdo-shellcheck)
==============================================================================
/home/vsts/work/_temp/shellcheck-latest/shellcheck-latest/shellcheck myscripts.sh
You are downloading ShellCheck from the wrong URL!
Please update to the latest one:
https://github.com/koalaman/shellcheck/releases/download/latest/shellcheck-latest.linux.x86_64.tar.xz

For more information, see:
https://github.com/koalaman/shellcheck/issues/1871
##[error]ShellCheck scan failed! Check the logs for violation details.
Finishing: Run ShellCheck
-->
```

<!-- 
```
##[section]Starting: Install ShellCheck
==============================================================================
Task         : Install ShellCheck
Description  : Installs Specified ShellCheck Version
Version      : 0.0.17
Author       : Swellaby
Help         : [More Information](https://github.com/swellaby/azdo-shellcheck)
==============================================================================
module.js:471
    throw err;
    ^

Error: Cannot find module 'semver'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/home/vsts/work/_tasks/install-shellcheck_3f74db91-b37c-4602-bb92-2658c6d136f2/0.0.17/node_modules/azure-pipelines-task-lib/internal.js:10:14)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
##[error]Exit code 1 returned from process: file name '/home/vsts/agents/2.148.2/externals/node/bin/node', arguments '"/home/vsts/work/_tasks/install-shellcheck_3f74db91-b37c-4602-bb92-2658c6d136f2/0.0.17/runner.js"'.
##[section]Finishing: Install ShellCheck
```
-->

--- a/src/tasks/install/installer.ts
+++ b/src/tasks/install/installer.ts
@@ -8,7 +8,7 @@ import toolRunner = require('azure-pipelines-task-lib/toolrunner');
 
 import ShellCheckVersion = require('./shellcheck-version');
 
-const shellCheckBinaryUrlBase = 'https://github.com/koalaman/shellcheck/releases/download'
+const shellCheckBinaryUrlBase = 'https://github.com/koalaman/shellcheck/releases/download';
 
 /**
  * @private
@@ -60,7 +60,8 @@ const installForWindows = async (version: string) => {
     const zipName = `shellcheck-${version}.zip`;
     const zipDownloadUrl = `${shellCheckBinaryUrlBase}/${version}/${zipName}`;
     const zipDownloadLocation = await toolLib.downloadTool(zipDownloadUrl, zipName);
-    const zipExtractionLocation = await toolLib.extractZip(zipDownloadLocation)
+    const zipExtractionLocation = await toolLib.extractZip(zipDownloadLocation);
+    taskLib.mv(`${zipExtractionLocation}/shellcheck-${version}.exe`, `${zipExtractionLocation}/shellcheck.exe`);
     toolLib.prependPath(zipExtractionLocation);
 };
 

--- a/src/tasks/install/installer.ts
+++ b/src/tasks/install/installer.ts
@@ -8,14 +8,15 @@ import toolRunner = require('azure-pipelines-task-lib/toolrunner');
 
 import ShellCheckVersion = require('./shellcheck-version');
 
-const shellCheckBinaryUrlBase = 'https://shellcheck.storage.googleapis.com';
+const shellCheckBinaryUrlBase = 'https://github.com/koalaman/shellcheck/releases/download'
 
 /**
  * @private
  */
 const installForLinux = async (version: string) => {
     const architecture = os.arch();
-    let tarballName;
+    let tarballName = '';
+    // let baseUrl = '';
     if (architecture === 'x64') {
         tarballName = `shellcheck-${version}.linux.x86_64.tar.xz`;
     } else if (architecture === 'arm64') {
@@ -24,7 +25,7 @@ const installForLinux = async (version: string) => {
         throw new Error(`Unsupported architecture ${architecture}`);
     }
 
-    const downloadUrl = `${shellCheckBinaryUrlBase}/${tarballName}`;
+    const downloadUrl = `${shellCheckBinaryUrlBase}/${version}/${tarballName}`;
     const tarballLocation = await toolLib.downloadTool(downloadUrl);
     // The ShellCheck binary tarballs are not gzip compressed. The toolLib.extractTar
     // function always adds the 'z' tar option which would always fail for ShellCheck.
@@ -56,9 +57,11 @@ const installForMac = async (version: string) => {
  * @private
  */
 const installForWindows = async (version: string) => {
-    const downloadUrl = `${shellCheckBinaryUrlBase}/shellcheck-${version}.exe`;
-    const downloadLocation = await toolLib.downloadTool(downloadUrl, 'shellcheck.exe');
-    toolLib.prependPath(path.parse(downloadLocation).dir);
+    const zipName = `shellcheck-${version}.zip`;
+    const zipDownloadUrl = `${shellCheckBinaryUrlBase}/${version}/${zipName}`;
+    const zipDownloadLocation = await toolLib.downloadTool(zipDownloadUrl, zipName);
+    const zipExtractionLocation = await toolLib.extractZip(zipDownloadLocation)
+    toolLib.prependPath(zipExtractionLocation);
 };
 
 /**

--- a/src/tasks/install/task.json
+++ b/src/tasks/install/task.json
@@ -8,8 +8,8 @@
   "helpMarkDown": "[More Information](https://github.com/swellaby/azdo-shellcheck)",
   "version": {
     "Major": 0,
-    "Minor": 1,
-    "Patch": 52
+    "Minor": 2,
+    "Patch": 0
   },
   "instanceNameFormat": "Install ShellCheck",
   "inputs": [

--- a/test/unit/tasks/install/installer.ts
+++ b/test/unit/tasks/install/installer.ts
@@ -25,8 +25,8 @@ suite('installers', () => {
         arg: (_val) => null,
         exec: (_options) => null
     };
-    // const shellCheckBinaryUrlBase = 'https://shellcheck.storage.googleapis.com';
-    const shellCheckBinaryUrlBase = 'https://github.com/koalaman/shellcheck/releases/download'
+
+    const shellCheckBinaryUrlBase = 'https://github.com/koalaman/shellcheck/releases/download';
 
     setup(() => {
         osTypeStub = Sinon.stub(os, 'type');
@@ -199,25 +199,31 @@ suite('installers', () => {
     });
 
     suite('Windows installer', () => {
-        const version = '0.5.0';
-        const downloadFileName = `shellcheck-${version}.exe`;
-        const shellcheckFileName = 'shellcheck.exe';
+        const version = 'v0.5.0';
+        const downloadFileName = `shellcheck-${version}.zip`;
         const downloadDirectory = 'c:/users/me/temp';
-        const downloadPath = `${downloadDirectory}/${shellcheckFileName}`;
-        let pathParseStub: Sinon.SinonStub;
+        const downloadPath = `${downloadDirectory}/${downloadFileName}`;
+        const extractionDirectory = `c:/agent/_work/tool/shellcheck/${version}`;
+        let taskLibMvStub: Sinon.SinonStub;
+        let toolLibExtractZipStub: Sinon.SinonStub;
 
         setup(() => {
             osTypeStub.callsFake(() => 'Windows_NT');
             toolLibDownloadStub.callsFake(() => Promise.resolve(downloadPath));
-            pathParseStub = Sinon.stub(path, 'parse');
-            pathParseStub.withArgs(downloadPath).callsFake(() => ({ dir: downloadDirectory }));
+            toolLibExtractZipStub = Sinon.stub(toolLib, 'extractZip');
+            toolLibExtractZipStub.withArgs(downloadPath).callsFake(() => Promise.resolve(extractionDirectory));
+            taskLibMvStub = Sinon.stub(taskLib, 'mv');
         });
 
         test('Should install correctly', async () => {
-            const expectedDownloadUrl = `${shellCheckBinaryUrlBase}/${downloadFileName}`;
+            const expectedDownloadUrl = `${shellCheckBinaryUrlBase}/${version}/${downloadFileName}`;
             await installer.installShellCheck(version);
-            assert.isTrue(toolLibDownloadStub.calledWithExactly(expectedDownloadUrl, shellcheckFileName));
-            assert.isTrue(toolLibPrependPathStub.calledWithExactly(downloadDirectory));
+            assert.isTrue(toolLibDownloadStub.calledWithExactly(expectedDownloadUrl, downloadFileName));
+            assert.isTrue(toolLibPrependPathStub.calledWithExactly(extractionDirectory));
+            assert.isTrue(taskLibMvStub.calledWithExactly(
+                `${extractionDirectory}/shellcheck-${version}.exe`,
+                `${extractionDirectory}/shellcheck.exe`
+            ));
         });
 
         test('Should bubble errors', async () => {

--- a/test/unit/tasks/install/installer.ts
+++ b/test/unit/tasks/install/installer.ts
@@ -25,7 +25,8 @@ suite('installers', () => {
         arg: (_val) => null,
         exec: (_options) => null
     };
-    const shellCheckBinaryUrlBase = 'https://shellcheck.storage.googleapis.com';
+    // const shellCheckBinaryUrlBase = 'https://shellcheck.storage.googleapis.com';
+    const shellCheckBinaryUrlBase = 'https://github.com/koalaman/shellcheck/releases/download'
 
     setup(() => {
         osTypeStub = Sinon.stub(os, 'type');
@@ -54,7 +55,7 @@ suite('installers', () => {
     });
 
     suite('Linux installer', () => {
-        const version = '0.6.0';
+        const version = 'v0.6.0';
         const binaryDirectoryName = `shellcheck-${version}`;
         const downloadDirectory = '/foo/bar';
         const tempDirectoryPath = '/vsts/work/_temp';
@@ -92,7 +93,7 @@ suite('installers', () => {
 
         test('Should install correctly on 64-bit architecture', async () => {
             osArchStub.callsFake(() => 'x64');
-            const expectedDownloadUrl = `${shellCheckBinaryUrlBase}/shellcheck-${version}.linux.x86_64.tar.xz`;
+            const expectedDownloadUrl = `${shellCheckBinaryUrlBase}/${version}/shellcheck-${version}.linux.x86_64.tar.xz`;
             await installer.installShellCheck(version);
             assert.isTrue(toolLibDownloadStub.calledWithExactly(expectedDownloadUrl));
             assert.isTrue(taskLibGetVariableStub.calledWithExactly('Agent.TempDirectory'));
@@ -105,7 +106,7 @@ suite('installers', () => {
 
         test('Should install correctly on arm 64-bit architecture', async () => {
             osArchStub.callsFake(() => 'arm64');
-            const expectedDownloadUrl = `${shellCheckBinaryUrlBase}/shellcheck-${version}.linux.armv6hf.tar.xz`;
+            const expectedDownloadUrl = `${shellCheckBinaryUrlBase}/${version}/shellcheck-${version}.linux.armv6hf.tar.xz`;
             await installer.installShellCheck(version);
             assert.isTrue(toolLibDownloadStub.calledWithExactly(expectedDownloadUrl));
             assert.isTrue(taskLibGetVariableStub.calledWithExactly('Agent.TempDirectory'));

